File: C/Programs/NumPalindrome.c
//C Program to check whether a number is palindrome or not

#include <stdio.h>
int main() {
  int num, reversed = 0, remainder, original;
    printf("Enter a number: ");
    scanf("%d", &num);
    original = num;

    // logic to reverse the input number
    while (num != 0) {
        remainder = num % 10;
        reversed = reversed * 10 + remainder;
        num /= 10;
    }

    // printing palindrome if orignal == reversed
    if (original == reversed)
        printf("%d is a palindrome.", original);
    else
        printf("%d is not a palindrome.", original);

    return 0;
}

File: Java/Programs/NumPalindrome.java
	import java.util.*;
	public class NumPalindrome {

		public static void main(String[] args){
			int num, reversed=0, remainder, original_num;
			Scanner scan=new Scanner(System.in);
			System.out.print("Enter a number: ");
			num = scan.nextInt(); //getting num
			original_num = num;
			
			scan.close();
			
			while (num != 0){
				remainder = num % 10;
				reversed = (reversed * 10) + remainder;
				num /= 10;
			}
			
			if(reversed == original_num)
				System.out.println(original_num+ " is palindrome number"); //Printing num as palindrome
			else
			    System.out.println(original_num+ " is not a palindrome number"); //Printing num is not palindrome
		}
	}